ui/src/ducks/inventory/index.js

```diff
@@ -1,17 +1,89 @@
+import axios from 'axios'
 import { createAction, handleActions } from 'redux-actions'
 
 const actions = {
   INVENTORY_GET_ALL: 'inventory/get_all',
+  INVENTORY_GET_ALL_PENDING: 'inventory/get_all_PENDING',
+  INVENTORY_SAVE: 'inventory/save',
+  INVENTORY_DELETE: 'inventory/delete',
+  INVENTORY_REFRESH: 'inventory/refresh',
+  INVETORY_GET_BY_ID: 'inventory/get_by_id',
+  INVENTORY_UPDATE: 'inventory/update'
 }
 
 export let defaultState = {
-  all: []
+  all: [],
+  fetched: false,
 }
 
-export const findInventory = createAction(actions.INVENTORY_GET_ALL, () => {
+export const findInventory = createAction(actions.INVENTORY_GET_ALL, () =>
   //TODO
-})
+  (dispatch, getState, config) => axios
+    .get(`${config.restAPIUrl}/inventory`)
+    .then((suc) => dispatch(refreshInventory(suc.data)))
+)
+
+export const saveInventory = createAction(actions.INVENTORY_SAVE, (inventory) =>
+  (dispatch, getState, config) => axios
+    .post(`${config.restAPIUrl}/inventory`, inventory)
+    .then((suc) => {
+      const invs = []
+      getState().inventory.all.forEach(inv => {
```

Comment:
This will always be true because you set id to null in the dao

Reply:
Would they not have the mongoId to compare?